yb-voyager/cmd/importData.go

```diff
@@ -969,69 +969,102 @@ func splitFilesForTable(state *ImportDataState, filePath string, t sqlname.NameT
 
 	var readLineErr error = nil
 	var line string
+	var currentBytesRead int64
 	var batchWriter *BatchWriter
 	header := ""
 	if dataFileDescriptor.HasHeader {
 		header = dataFile.GetHeader()
 	}
-	for readLineErr == nil {
 
-		if batchWriter == nil {
-			batchWriter = state.NewBatchWriter(filePath, t, batchNum)
-			err := batchWriter.Init()
+	// Helper function to initialize a new batchWriter
+	initBatchWriter := func() {
+		batchWriter = state.NewBatchWriter(filePath, t, batchNum)
+		err := batchWriter.Init()
+		if err != nil {
+			utils.ErrExit("initializing batch writer for table %q: %s", t, err)
+		}
+		// Write the header if necessary
+		if header != "" && dataFileDescriptor.FileFormat == datafile.CSV {
+			err = batchWriter.WriteHeader(header)
 			if err != nil {
-				utils.ErrExit("initializing batch writer for table %q: %s", t, err)
-			}
-			if header != "" && dataFileDescriptor.FileFormat == datafile.CSV {
-				err = batchWriter.WriteHeader(header)
-				if err != nil {
-					utils.ErrExit("writing header for table %q: %s", t, err)
-				}
+				utils.ErrExit("writing header for table %q: %s", t, err)
 			}
 		}
+	}
+
+	// Function to finalize and submit the current batch
+	finalizeBatch := func(isLastBatch bool, offsetEnd int64, bytesInBatch int64) {
+		batch, err := batchWriter.Done(isLastBatch, offsetEnd, bytesInBatch)
+		if err != nil {
+			utils.ErrExit("finalizing batch %d: %s", batchNum, err)
+		}
+		batchWriter = nil
+		submitBatch(batch, updateProgressFn, importBatchArgsProto)
+
+		// Increment batchNum only if this is not the last batch
+		if !isLastBatch {
+			batchNum++
+		}
+	}
 
-		line, readLineErr = dataFile.NextLine()
+	for readLineErr == nil {
+
+		if batchWriter == nil {
+			initBatchWriter() // Create a new batchWriter
+		}
+
+		line, currentBytesRead, readLineErr = dataFile.NextLine()
 		if readLineErr == nil || (readLineErr == io.EOF && line != "") {
 			// handling possible case: last dataline(i.e. EOF) but no newline char at the end
 			numLinesTaken += 1
 		}
+		log.Debugf("Batch %d: totalBytesRead %d, currentBytes %d \n", batchNum, dataFile.GetBytesRead(), currentBytesRead)
+		if currentBytesRead > tdb.MaxBatchSizeInBytes() {
+			//If a row is itself larger than MaxBatchSizeInBytes erroring out
+			ybSpecificMsg := ""
+			if tconf.TargetDBType == YUGABYTEDB {
+				ybSpecificMsg = ", but should be strictly lower than the the rpc_max_message_size on YugabyteDB (default 267386880 bytes)"
```

Comment:
nit: define a constant for `267386880`

Reply:
I don't see a point for this as constant as we are not going to use it anywhere, it's just for printing.